src/components/tx/SignOrExecuteForm/SignOrExecuteForm.test.tsx

```diff
@@ -368,6 +376,114 @@ describe('SignOrExecuteForm', () => {
     expect(result.getByText('Estimating...')).toBeDisabled()
   })
 
+  it('relays a 1 out of 2 signed transaction', async () => {
+    const relaySpy = jest.fn()
+    const proposeSpy = jest.fn(() => Promise.resolve({ txId: '0xdead' }))
+    jest.spyOn(txSender, 'default').mockImplementation(
+      () =>
+        ({
+          dispatchTxSigning: jest.fn(() => Promise.resolve({})),
+          dispatchTxProposal: proposeSpy,
+          dispatchTxRelay: relaySpy,
+        } as unknown as NullableTxSenderFunctions),
+    )
+
+    const mockTx = createSafeTx()
+
+    mockTx.addSignature({
+      signer: '0x123',
+      data: '0xEEE',
+      staticPart: () => '0xEEE',
+      dynamicPart: () => '',
+    })
+
+    const result = render(<SignOrExecuteForm isExecutable={true} onSubmit={jest.fn} safeTx={mockTx} txId="0xdead" />)
+
+    const submitButton = result.getByText('Submit')
+
+    act(() => {
+      fireEvent.click(submitButton)
+    })
+
+    await waitFor(() => {
+      expect(proposeSpy).toHaveBeenCalledTimes(1)
+      expect(relaySpy).toHaveBeenCalledTimes(1)
+    })
+  })
+
+  it('should not relay a not fully signed transaction with a connected SC wallet', async () => {
+    const relaySpy = jest.fn()
+    jest.spyOn(txSender, 'default').mockImplementation(
+      () =>
+        ({
+          dispatchTxProposal: jest.fn(() => Promise.resolve({})),
+          dispatchTxRelay: relaySpy,
+        } as unknown as NullableTxSenderFunctions),
+    )
+
+    // SC wallet connected
+    jest.spyOn(useIsSmartContractWallet, 'default').mockReturnValue([true, , false])
+
+    const mockTx = createSafeTx()
+
+    mockTx.addSignature({
+      signer: '0x123',
+      data: '0xEEE',
+      staticPart: () => '0xEEE',
+      dynamicPart: () => '',
+    })
+
+    const result = render(<SignOrExecuteForm isExecutable={true} onSubmit={jest.fn} safeTx={mockTx} txId="0xdead" />)
+
+    const submitButton = result.getByText('Submit')
+
+    act(() => {
+      fireEvent.click(submitButton)
+    })
+
+    await waitFor(() => expect(relaySpy).toHaveBeenCalledTimes(0))
+  })
+
+  it('relays a fully signed transaction', async () => {
+    const relaySpy = jest.fn()
+    jest.spyOn(txSender, 'default').mockImplementation(
+      () =>
+        ({
+          dispatchTxProposal: jest.fn(() => Promise.resolve({})),
+          dispatchTxRelay: relaySpy,
+        } as unknown as NullableTxSenderFunctions),
+    )
+
+    // SC wallet connected
+    jest.spyOn(useIsSmartContractWallet, 'default').mockReturnValue([true, , false])
```

Comment:
This is only testing the relay of fully signed transactions *from smart contract wallets, right? If so, can we adjust the description and add a test for normal wallets?